app/helpers/filter_participatory_spaces_helper.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module FilterParticipatorySpacesHelper
+  def permissions(participatory_spaces)
+    user_auth = Decidim::Authorization.find_by(decidim_user_id: current_user)
+    user_permissions = user_auth.present? && user_auth.metadata.present? ? user_auth.metadata : {}
+
+    without_permissions(participatory_spaces) + filter_permissions(participatory_spaces, user_permissions)
+  end
+
+  def without_permissions(participatory_spaces)
+    participatory_spaces.select { |space| query_permissions(space) }
+  end
+
+  def filter_permissions(participatory_spaces, user_permissions)
+    with_permissions = participatory_spaces.reject { |space| query_permissions(space) }
+
+    if with_permissions.present?
+      with_permissions = with_permissions.reject { |space| Hash[*(space.suara_permissions.map { |sp| sp } & user_permissions&.map { |us| us }).flatten].values.all?(&:blank?) }
```

Comment:
is `space.suara_permissions.map { |sp| sp }` the same as `space.suara_permissions.to_a` or `space.suara_permissions.entries`?

Reply:
Can you explain the magic in here: `Hash[*(space.suara_permissions.map { |sp| sp } & user_permissions&.map { |us| us }).flatten].values.all?`

Reply:
I have changed the operation a little bit because there was a case that I didn't contemplate.

The first condition `Hash[*(space.suara_permissions.to_a & user_permissions&.to_a).flatten].values.all?(&:blank?)` take the space and user permissions and check the common values. If none of them match, discard that participatory space.

The second condition `Hash[*(space.suara_permissions.to_a - user_permissions&.to_a).flatten].values.reject(&:empty?).any?` take the space and user permissions values that are different. If there is a different value, discard that participatory space.

If you think there is any way to do it better, please tell me and I will learn :)

Reply:
I improve with `<=` hash method :)